### PR TITLE
Oauth testing

### PR DESCRIPTION
**Pull request by brian-greeson**
# Oauth Testing 
## Added
 - Change signout route to sessions controller
 - Add oauth configure to rails_helper
 - Add feature test for connecting github with oauth
 - Remove unused destroy action in Github_Sessions controller

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -37,7 +37,8 @@
   get '/video', to: 'video#show'
 
   get "/auth/:provider/callback" => "github_sessions#create"
-  get "/signout" => "github_sessions#destroy", :as => :signout
+  get "/auth/failure", to: "github_sessions#create"
+  get "/signout" => "sessions_controller#destroy", :as => :signout
```
Comment (brian-greeson) [inline config/routes.rb:41]: Signout now uses the sessions_controller

File: app/controllers/github_sessions_controller.rb [hunk 2]
```diff
@@ -1,12 +1,12 @@
 class GithubSessionsController < ApplicationController
   def create
     auth = request.env["omniauth.auth"]
-    current_user.update!(github_token: auth[:credentials][:token], github_username: auth[:extra][:raw_info][:login])
-    redirect_to root_url, :notice => "Signed in with Github!"
+    if auth
```
Comment (brian-greeson) [inline app/controllers/github_sessions_controller.rb:4]: Added Sad paths because paths are sad

File: config/routes.rb (hunk 1 above)
Comment (StephanieFriend) [reply config/routes.rb:41]: Awesome!

File: spec/features/user/user_can_connect_to_github_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+    visit dashboard_path
+    
+    click_on "Connect To Github"
+
+    expect(current_path).to eq(dashboard_path)
+    expect(page).to have_content("Failed to connect to Github")
+  end
+
+end
```
Comment (StephanieFriend) [inline spec/features/user/user_can_connect_to_github_spec.rb:37]: Awesome looking test!

**Review by StephanieFriend — APPROVED** [review]
OAuth test looks great!

**Pull request merged**